processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -67,4 +73,80 @@ public static Filter toFilter(DimFilter dimFilter)
   {
     return dimFilter == null ? null : dimFilter.toFilter();
   }
+
+  /**
+   * Return the union of bitmaps for all values matching a particular predicate.
+   *
+   * @param dimension dimension to look at
+   * @param selector  bitmap selector
+   * @param predicate predicate to use
+   *
+   * @return bitmap of matching rows
+   */
+  public static ImmutableBitmap matchPredicate(
+      final String dimension,
+      final BitmapIndexSelector selector,
+      final Predicate<String> predicate
+  )
+  {
+    // Null predicate -> nothing matches
+    if (predicate == null) {
+      return selector.getBitmapFactory().makeEmptyImmutableBitmap();
```

Comment:
Actually now that I look at usages, nothing actually ever passes in a null predicate. So I am gonna remove this code and add a null-check for `predicate`.